anoncreds/src/services/verifier.rs

```diff
@@ -788,11 +796,13 @@ fn process_filter(
     );
     match tag {
         tag_ @ "schema_id" => precess_filed(tag_, filter.schema_id.to_string(), tag_value),
-        tag_ @ "schema_issuer_did" => precess_filed(tag_, &filter.schema_issuer_did, tag_value),
+        tag_ @ "schema_issuer_did" => {
+            precess_filed(tag_, filter.schema_issuer_id.to_owned(), tag_value)
+        }
         tag_ @ "schema_name" => precess_filed(tag_, &filter.schema_name, tag_value),
         tag_ @ "schema_version" => precess_filed(tag_, &filter.schema_version, tag_value),
         tag_ @ "cred_def_id" => precess_filed(tag_, &filter.cred_def_id.to_string(), tag_value),
-        tag_ @ "issuer_did" => precess_filed(tag_, &filter.issuer_did, tag_value),
+        tag_ @ "issuer_did" => precess_filed(tag_, filter.issuer_id.to_owned(), tag_value),
```

Comment:
I see you left the keys as `_did`. We need to think a bit about backwards interop here. We probably want `_did` to be valid, but only for legacy (unqualified identifiers). For all others we need to use the `_id` variant.

This way we keep the old exchange valid

Reply:
OR we require the input to be `_id` and the user of the library must manually transform it from `_did` to `_id`. But I think it's maybe easier to support both in here?

Reply:
Hmm yeah I am not too sure. I did not notice the key, so that is a good catch. 

Thinking whether this would work okay:

```rust
        tag_ @ "issuer_did" => precess_filed("issuer_id", filter.issuer_id.to_owned(), tag_value),
        tag_ @ "issuer_id" => precess_filed(tag_, filter.issuer_id.to_owned(), tag_value),

```

Reply:
Hmm not sure if we should mix them. Where is this code used for? Is it only for internal processing? We don't want to modify the output model. And they probably shouldn't be allowed to be defined at the same time?

Reply:
Well, I'd really like to make the old indy flow (pre standardization) still valid. If we remove the `_did` part we break interop. Until now, all breaking changes have been on the ledger models, which you resolve on your own. So if I support new AnonCreds, and you support old anoncreds we're all good (as long as I also use old).

Reply:
Maybe something to discuss during the WG call. It has quite some impact on interop. Maybe we keep both in the spec, but we make the anoncreds lib only support, `_id`. But that wouldn't make a lot of sense also...

Reply:
Another thing to note is we added validation now to the `issuer_id`. So if we were to reuse it with a legacy identifier, it would not work (as validation will fail).

Reply:
Legacy indy identifiers will still be valid for dids, meaning the issuer_id won't be always an uri

Reply:
Okay then I need to redo the validation for the identifiers as it can be a URI or the legacy did indy.

Would this suffice:

```
1. Check if URI
2. If URI, success
3. If not URI, check legacy indy identifier
  - Check length (not sure which lengths, I thought there were 2 different lengths)
  - Check if all chars are in the base58 alphabet
```

Reply:
Yes. I think the length is 21 or 22 (we have a regex in AFJ)

Reply:
Awesome. Will add that to this PR.

Reply:
It should be fixed with the latest commit.